### PR TITLE
Add simdutf support for parsing network messages

### PR DESCRIPTION
**Pull request by ranisalt**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Uses simdutf to perform utf8<->latin1 conversion for network messages. Benchmark incoming.

**Issues addressed:** 

**How to test:** 

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: vcpkg.json [hunk 1]
```diff
@@ -48,17 +53,18 @@
...
       "dependencies": [
         "boost-test"
       ]
     }
   },
-  "default-features": [
-    "lua",
-    "http",
-    "libmariadb"
-  ],
```
Comment (ranisalt) [inline vcpkg.json:62]: I have to remove this because it is impossible to _disable_ a feature from CMake, so we can't choose one or the other, only add to the default feature set.

**Comment by ranisalt** [comment]
simdutf is so ridiculously faster that I think we should just drop boost::locale here. It's used by Node, WebKit, Chromium, Cloudflare and regularly updated.

<details>
<summary>Benchmark</summary>

Results:

```
----------------------------------------------------------------------
Benchmark                            Time             CPU   Iterations
----------------------------------------------------------------------
Boost_Latin1_to_UTF8/16            264 ns          264 ns      2677449
Boost_Latin1_to_UTF8/64            622 ns          622 ns      1131405
Boost_Latin1_to_UTF8/256          2049 ns         2048 ns       339282
Boost_Latin1_to_UTF8/1024         8151 ns         8145 ns        84807
Boost_Latin1_to_UTF8/4096        37328 ns        37299 ns        18881
SimdUTF_Latin1_to_UTF8/16         12.9 ns         12.9 ns     50538824
SimdUTF_Latin1_to_UTF8/64         14.4 ns         14.4 ns     48951326
SimdUTF_Latin1_to_UTF8/256        22.4 ns         22.4 ns     30997593
SimdUTF_Latin1_to_UTF8/1024       47.1 ns         47.0 ns     14955163
SimdUTF_Latin1_to_UTF8/4096        130 ns          130 ns      5424932
Boost_UTF8_to_Latin1/16            151 ns          151 ns      4610094
Boost_UTF8_to_Latin1/64            181 ns          180 ns      3880563
Boost_UTF8_to_Latin1/256           281 ns          281 ns      2472421
Boost_UTF8_to_Latin1/1024          703 ns          703 ns       983878
Boost_UTF8_to_Latin1/4096         3299 ns         3296 ns       213637
SimdUTF_UTF8_to_Latin1/16         2.19 ns         2.19 ns    323556710
SimdUTF_UTF8_to_Latin1/64         2.50 ns         2.50 ns    277252101
SimdUTF_UTF8_to_Latin1/256        4.88 ns         4.87 ns    139540796
SimdUTF_UTF8_to_Latin1/1024       14.6 ns         14.6 ns     48851232
SimdUTF_UTF8_to_Latin1/4096       71.4 ns         71.4 ns      9801352
```

Code:

```cpp
#include <benchmark/benchmark.h>
#include <boost/locale.hpp>
#include <random>
#include <simdutf.h>

// Generate random Latin1 input string
std::string generate_latin1_data(size_t size)
{
	std::string result(size, '\0');
	std::mt19937 gen(123);
	std::uniform_int_distribution<> dist(0x00, 0xFF);

	for (auto& ch : result) ch = static_cast<char>(dist(gen));

	return result;
}

static void Boost_Latin1_to_UTF8(benchmark::State& state)
{
	const std::string latin1_data = generate_latin1_data(state.range(0));

	for (auto _ : state) {
		auto utf8 = boost::locale::conv::from_utf(latin1_data, "ISO-8859-1");
		benchmark::DoNotOptimize(utf8);
	}
}
BENCHMARK(Boost_Latin1_to_UTF8)->RangeMultiplier(4)->Range(16, 4096);

static void SimdUTF_Latin1_to_UTF8(benchmark::State& state)
{
	const std::string latin1_data = generate_latin1_data(state.range(0));
	auto stringLen = simdutf::utf8_length_from_latin1(latin1_data.data(), latin1_data.size());

	std::string output(stringLen, '\0');
	for (auto _ : state) {
		size_t len =
		    simdutf::convert_latin1_to_utf8_safe(latin1_data.data(), latin1_data.size(), output.data(), stringLen);
		benchmark::DoNotOptimize(len);
	}
}
BENCHMARK(SimdUTF_Latin1_to_UTF8)->RangeMultiplier(4)->Range(16, 4096);

static void Boost_UTF8_to_Latin1(benchmark::State& state)
{
	const std::string latin1_data = generate_latin1_data(state.range(0));

	const auto utf8 = boost::locale::conv::from_utf(latin1_data, "ISO-8859-1");
	for (auto _ : state) {
		auto latin1 = boost::locale::conv::to_utf<char>(utf8, "ISO-8859-1");
		benchmark::DoNotOptimize(latin1);
	}
}
BENCHMARK(Boost_UTF8_to_Latin1)->RangeMultiplier(4)->Range(16, 4096);

static void SimdUTF_UTF8_to_Latin1(benchmark::State& state)
{
	const std::string latin1_data = generate_latin1_data(state.range(0));
	std::vector<char> utf8_data;
	utf8_data.resize(latin1_data.size() * 2);

	auto stringLen = simdutf::latin1_length_from_utf8(utf8_data.data(), utf8_data.size());

	std::string output(stringLen, '\0');
	for (auto _ : state) {
		size_t len = simdutf::convert_utf8_to_latin1(utf8_data.data(), utf8_data.size(), output.data());
		benchmark::DoNotOptimize(len);
	}
}
BENCHMARK(SimdUTF_UTF8_to_Latin1)->RangeMultiplier(4)->Range(16, 4096);
```

Built with `g++ -DBENCHMARK_STATIC_DEFINE -DBOOST_LOCALE_NO_LIB -DBOOST_LOCALE_STATIC_LINK -O3 -DNDEBUG -std=c++20 -o utf utf.cpp`

</details>

simdutf is 18 to 287x (!) faster from latin1 to utf8 and 45 to 70x faster from utf8 to latin1. It depends on the processor SIMD support, just like XTEA before it.

File: src/networkmessage.cpp [hunk 2]
```diff
@@ -20,12 +26,16 @@ std::string NetworkMessage::getString(uint16_t stringLen /* = 0*/)
 		return {};
 	}
 
-	auto it = buffer.data() + info.position;
+	auto it = reinterpret_cast<char*>(buffer.data() + info.position);
 	info.position += stringLen;
```
[the remark this answers is not in the document]
Comment (ranisalt) [reply src/networkmessage.cpp:24]: Not for now

File: src/networkmessage.cpp [hunk 3]
```diff
@@ -20,12 +20,14 @@ std::string NetworkMessage::getString(uint16_t stringLen /* = 0*/)
...
 
-	std::string_view latin1Str{reinterpret_cast<char*>(it), stringLen};
-	return boost::locale::conv::to_utf<char>(latin1Str.data(), latin1Str.data() + latin1Str.size(), "ISO-8859-1",
-	                                         boost::locale::conv::skip);
+	auto out = std::string(simdutf::utf8_length_from_latin1(it, stringLen), '\0');
+	auto outLen = simdutf::convert_latin1_to_utf8(it, stringLen, out.data());
+	assert(outLen <= out.size());
+
+	return out.substr(0, outLen);
```
Comment (ranisalt) [inline src/networkmessage.cpp]: I will check if `utf8_length_from_latin1` is the exact length, I'm assuming it's just a worst case scenario but if it's always correct then I can drop this slicing

Comment (ranisalt) [reply src/networkmessage.cpp]: Update: yes, it's always exact

```cpp
#define BOOST_TEST_MODULE simdutf

#include "../otpch.h"

#include <boost/test/unit_test.hpp>
#include <simdutf.h>

// Helper: Generate a random Latin1 string of given length
std::string random_latin1_string(size_t length, std::mt19937& rng)
{
	std::uniform_int_distribution<> dist(0, 255);
	std::string result;
	result.reserve(length);
	for (size_t i = 0; i < length; ++i) {
		result.push_back(static_cast<char>(dist(rng)));
	}
	return result;
}

// Helper: Generate edge-case Latin1 strings (all 0x00, all 0xFF, alternating, etc.)
std::vector<std::string> generate_edge_case_latin1(size_t length)
{
	std::vector<std::string> cases;
	// All zeros
	cases.emplace_back(length, '\0');
	// All 0xFF
	cases.emplace_back(length, static_cast<char>(0xFF));
	// All 0x7F
	cases.emplace_back(length, static_cast<char>(0x7F));
	// Alternating 0x00 and 0xFF
	std::string alt(length, '\0');
	for (size_t i = 0; i < length; ++i) {
		alt[i] = (i % 2 == 0) ? static_cast<char>(0x00) : static_cast<char>(0xFF);
	}
	cases.push_back(alt);
	// Ascending bytes
	std::string asc(length, '\0');
	for (size_t i = 0; i < length; ++i) {
		asc[i] = static_cast<char>(i % 256);
	}
	cases.push_back(asc);
	// Descending bytes
	std::string desc(length, '\0');
	for (size_t i = 0; i < length; ++i) {
		desc[i] = static_cast<char>(255 - (i % 256));
	}
	cases.push_back(desc);

	return cases;
}

BOOST_AUTO_TEST_CASE(latin1_to_utf8_lengths_match)
{
	std::mt19937 rng(12345); // Fixed seed for reproducibility

	// Allocate output buffer (max 2x input size)
	std::array<char, static_cast<int>(std::numeric_limits<uint16_t>::max()) * 2 + 4> utf8buf;

	for (auto len = 0; len < std::numeric_limits<uint16_t>::max(); ++len) {
		if (len % 256 == 0) {
			// Print progress every 256 iterations
			std::cout << "Testing length: " << len << std::endl;
		}

		std::string latin1 = random_latin1_string(len, rng);

		size_t expected_len = simdutf::convert_latin1_to_utf8(latin1.data(), len, utf8buf.data());
		size_t computed_len = simdutf::utf8_length_from_latin1(latin1.data(), len);

		BOOST_CHECK_EQUAL(expected_len, computed_len);

		// Test edge cases
		auto edge_cases = generate_edge_case_latin1(len);
		for (const auto& latin1 : edge_cases) {
			size_t expected_len = simdutf::convert_latin1_to_utf8(latin1.data(), len, utf8buf.data());
			size_t computed_len = simdutf::utf8_length_from_latin1(latin1.data(), len);

			BOOST_CHECK_EQUAL(expected_len, computed_len);
		}
	}
}

BOOST_AUTO_TEST_CASE(utf8_to_latin1_lengths_match)
{
	std::mt19937 rng(54321);

	std::array<char, static_cast<int>(std::numeric_limits<uint16_t>::max()) + 4> latin1buf;
	std::array<char, static_cast<int>(std::numeric_limits<uint16_t>::max()) * 2 + 4> utf8buf;

	for (auto len = 0; len < std::numeric_limits<uint16_t>::max(); ++len) {
		if (len % 256 == 0) {
			// Print progress every 256 iterations
			std::cout << "Testing length: " << len << std::endl;
		}

		std::string latin1 = random_latin1_string(len, rng);

		// Encode to UTF-8 first
		size_t utf8len = simdutf::convert_latin1_to_utf8(latin1.data(), len, utf8buf.data());

		// Round-trip: decode UTF-8 back to Latin1
		size_t expected_len = simdutf::convert_utf8_to_latin1(utf8buf.data(), utf8len, latin1buf.data());
		size_t computed_len = simdutf::latin1_length_from_utf8(utf8buf.data(), utf8len);

		BOOST_CHECK_EQUAL(expected_len, computed_len);
	}
}
```

**Review by ramon-bernardo — APPROVED** [review]
🚀

**Comment by Codinablack** [comment]
Fantastic improvement IMO. LGTM
